### PR TITLE
Add centralized HPA api

### PR DESCRIPTION
**Pull request by jwcesign**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
* Add centralized HPA api

**Which issue(s) this PR fixes**:
Fixes #none

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by jwcesign** [comment]
cc @Poor12

File: pkg/apis/autoscaling/v1alpha1/hpaaggregator_types.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+// When the system load increases, it will query the metrics from multiple clusters and scales up the replicas.
+// When the system load decreases, it will query the metrics from multiple clusters and scales down the replicas.
+// After the replicas are scaled up/down, karmada-scheduler will schedule the replicas based on the policy.
+type HPAAggregator struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	// Spec is the specification of the HPAAggregator.
+	// +required
+	Spec HPAAggregatorSpec `json:"spec"`
```
Comment (Poor12) [inline pkg/apis/autoscaling/v1alpha1/hpaaggregator_types.go:34]: Here we do not use the original HPA spec is for extending new fields in the future, right?

Comment (jwcesign) [reply pkg/apis/autoscaling/v1alpha1/hpaaggregator_types.go:34]: Yes,

File: pkg/apis/autoscaling/v1alpha1/federatedhpa_types.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import (
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:shortName=fhpa,categories={karmada-io}
+
+// FederatedHPASpec is centralized HPA that can aggregate the metrics in multiple clusters.
```
Comment (RainbowMango) [inline pkg/apis/autoscaling/v1alpha1/federatedhpa_types.go:12]:
```suggestion
// FederatedHPA is centralized HPA that can aggregate the metrics in multiple clusters.
```

**Review by RainbowMango — COMMENTED** [review]
The lint is failing:
```
Error: pkg/apis/autoscaling/v1alpha1/zz_generated.register.go:45:3: cannot use &FederatedHPA{} (value of type *FederatedHPA) as "k8s.io/apimachinery/pkg/runtime".Object value in argument to scheme.AddKnownTypes: *FederatedHPA does not implement "k8s.io/apimachinery/pkg/runtime".Object (missing method DeepCopyObject)
Error: pkg/apis/autoscaling/v1alpha1/zz_generated.register.go:46:3: cannot use &FederatedHPAList{} (value of type *FederatedHPAList) as "k8s.io/apimachinery/pkg/runtime".Object value in argument to scheme.AddKnownTypes: *FederatedHPAList does not implement "k8s.io/apimachinery/pkg/runtime".Object (missing method DeepCopyObject) (typecheck)
```

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

File: pkg/apis/autoscaling/v1alpha1/federatedhpa_types.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+// FederatedHPA is centralized HPA that can aggregate the metrics in multiple clusters.
+// When the system load increases, it will query the metrics from multiple clusters and scales up the replicas.
+// When the system load decreases, it will query the metrics from multiple clusters and scales down the replicas.
+// After the replicas are scaled up/down, karmada-scheduler will schedule the replicas based on the policy.
+type FederatedHPA struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification of the HPAAggregator.
```
Comment (Poor12) [inline pkg/apis/autoscaling/v1alpha1/federatedhpa_types.go:20]: Some comments have not changed.
```suggestion
	// Spec is the specification of the FederatedHPA.
```

File: pkg/apis/autoscaling/v1alpha1/federatedhpa_types.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	// +required
+	Spec FederatedHPASpec `json:"spec"`
+
+	// Status is the current status of the HPAAggregator.
```
Comment (Poor12) [inline pkg/apis/autoscaling/v1alpha1/federatedhpa_types.go:24]: Ditto.

File: pkg/apis/autoscaling/v1alpha1/federatedhpa_types.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	// +optional
+	Status autoscalingv2.HorizontalPodAutoscalerStatus `json:"status"`
+}
+
+// FederatedHPASpec describes the desired functionality of the HPAAggregator.
```
Comment (Poor12) [inline pkg/apis/autoscaling/v1alpha1/federatedhpa_types.go:29]: Ditto.

**Pull request merged**